src/datapoint.jl

```diff
@@ -0,0 +1,60 @@
+## New Type: DataPoint
+# Struct for carrying statistic and systematic uncertainty of a DataPoint
+struct DataPoint{T<:Number} <: Number
+    val::T
+    stat_err::Tuple{T,T}
+    sys_err::Tuple{T,T}
+end
+
+# Helper functions
+_to_tuple(x::T where T<:Number) = (x,x)
+_to_tuple(x::Tuple) = promote(x...)
+
+## Constructors
+# default constructor
+DataPoint() = DataPoint(NaN, (NaN, NaN), (NaN, NaN))
+# dispatch constructor
+DataPoint(x::Number, stat, sys = zero(x)) = DataPoint(x, stat = stat, sys = sys)
+# keyword constructor for integers (ceils uncertainties)
+function DataPoint(x::Integer; stat=zero(x), sys=zero(x))
+    @assert all(real.((stat...,sys...)) .>= 0) "Only use positive uncertainties!"
```

Comment:
This does not catch the case where there is also an error for a possible the imaginary part. Consider as an example the case of `val=1` and `stat= 0 - 1*im` which currently does not error even though the imaginary part of `stat` is negative. We can just check if the errors are in the first quadrant and also create an internal function for that for clarity. 
```julia
_check_error_positivity(err) = all(x -> (π/2 >= angle(x)>=0), err)
```
and then replace the check in function by
```julia
@assert _check_error_positivity(stat) && _check_error_positivity(sys)
```
and add also a test of this function.

Reply:
Sounds good to me.